k4Reco/GaudiLumiCalClusterer/src/LumiCalClusterer.cpp

```diff
@@ -0,0 +1,218 @@
+#if _GENERAL_CLUSTERER_DEBUG == 1
+  info() << std::endl << "Global parameters for LumiCalClustererClass:" << std::endl;
+  info() << " m_cellRMax: " << m_cellRMax << std::endl
+         << " m_cellPhiMax: " << m_cellPhiMax << std::endl
+         << " m_zFirstLayer: " << m_zFirstLayer << std::endl
+         << " m_zLayerThickness: " << m_zLayerThickness << std::endl
+         << " m_zLayerPhiOffset[deg]: " << m_zLayerPhiOffset * 180. / M_PI << std::endl
+         << " m_rMin: " << m_rMin << std::endl
+         << " m_rMax: " << m_rMax << std::endl
+         << " m_rCellLength [mm]: " << m_rCellLength << std::endl
+         << " m_phiCellLength [rad]:" << m_phiCellLength << std::endl
+         << " m_methodCM: " << m_methodCM << std::endl
+         << " m_logWeightConst: " << m_logWeightConst << std::endl
+         << " m_elementsPercentInShowerPeakLayer: " << m_elementsPercentInShowerPeakLayer << std::endl
+         << " m_moliereRadius: " << m_moliereRadius << std::endl
+         << " m_minSeparationDistance: " << m_minSeparationDistance << std::endl
+         << " m_minClusterEngy - GeV: " << m_minClusterEngyGeV << std::endl
+         << " m_hitMinEnergy: " << m_hitMinEnergy << std::endl
+         << " m_thetaContainmentBounds[0]: " << m_thetaContainmentBounds[0] << std::endl
+         << " m_thetaContainmentBounds[1]: " << m_thetaContainmentBounds[1] << std::endl
+         << " m_middleEnergyHitBoundFrac: " << m_middleEnergyHitBoundFrac << std::endl
+         << " Clustering Options : " << std::endl
+         << "           _CLUSTER_MIDDLE_RANGE_ENGY_HITS    " << _CLUSTER_MIDDLE_RANGE_ENGY_HITS << std::endl
+         << "           _MOLIERE_RADIUS_CORRECTIONS        " << _MOLIERE_RADIUS_CORRECTIONS << std::endl
+         << "           _CLUSTER_MIXING_ENERGY_CORRECTIONS " << _CLUSTER_MIXING_ENERGY_CORRECTIONS << std::endl
+         << std::endl;
+#endif
+}
+
+/* ============================================================================
+   main actions in each event:
+   ========================================================================= */
+std::pair<RETVAL, edm4hep::CalorimeterHitCollection>
+LumiCalClustererClass::processEvent(edm4hep::SimCalorimeterHitCollection& col) {
+  // increment / initialize global variables
+  m_totEngyArm[-1] = m_totEngyArm[1] = 0.;
+  m_numHitsInArm[-1] = m_numHitsInArm[1] = 0;
+
+  MapIntMapIntVCalHit calHits;
+  MapIntMapIntLCCluster superClusterCM;
+
+  MapIntMapIntCalHit calHitsCellIdGlobal;
+
+  m_superClusterIdToCellId.clear();
+  m_superClusterIdToCellEngy.clear();
+  m_superClusterIdClusterInfo.clear();
+
+  /* --------------------------------------------------------------------------
+     Loop over all hits in the LCCollection and write the hits into std::vectors
+     of IMPL::CalorimeterHitImpl. Hits are split in two std::vectors, one for each arm
+     of LumiCal.
+     -------------------------------------------------------------------------- */
```

Comment:
This is too detailed and LCIO related to stay, IMO.

Reply:
I removed that one and other comments